### PR TITLE
fix(across-relayer): Key BridgePool.l2Tokens by chain ID

### PR DESCRIPTION
**Pull request by nicholaspai**
**Motivation**
Currently we assume there is one L2 token per BridgePool but there can be many depending on how many L2 tokens there are.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [X]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export interface InstantRelay {
 
 export interface BridgePoolData {
   contract: BridgePoolWeb3;
-  l2Token: string;
+  l2Token: { [key: string]: string }; // chainID=>L1TokenAddress
```
Comment (chrismaree) [inline packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts:55]:
```suggestion
  l2Token: { [key: string]: string }; // chainID=>L2TokenAddress
```

File: packages/insured-bridge-relayer/src/CrossDomainFinalizer.ts [hunk 2]
```diff
@@ -112,7 +112,7 @@ export class CrossDomainFinalizer {
 
   // Fetch L1 pool reserves for a given l2Token.
   private async _getL1PoolReserves(l2Token: string): Promise<BN> {
-    const bridgePool = this.l1Client.getBridgePoolForL2Token(l2Token).contract;
+    const bridgePool = this.l1Client.getBridgePoolForL2Token(l2Token, this.l2Client.chainId.toString()).contract;
```
Comment (chrismaree) [inline packages/insured-bridge-relayer/src/CrossDomainFinalizer.ts:115]: nice. I'm pretty sure this is the only place we use this method.

**Review by chrismaree — APPROVED** [review]
good fix.

File: packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts [hunk 3]
```diff
@@ -52,7 +52,7 @@ export interface InstantRelay {
 
 export interface BridgePoolData {
   contract: BridgePoolWeb3;
-  l2Token: string;
+  l2Token: { [key: string]: string }; // chainID=>L2TokenAddress
```
Comment (mrice32) [inline packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts:55]:
```suggestion
  l2Token: { [chainId: string]: string }; // chainID=>L2TokenAddress
```

File: packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts [hunk 4]
```diff
@@ -213,9 +213,11 @@ export class InsuredBridgeL1Client {
     return this.bridgePools[l1Token];
   }
 
-  getBridgePoolForL2Token(l2Token: string): BridgePoolData {
-    const bridgePoolData = Object.values(this.bridgePools).filter((bridgePool) => bridgePool.l2Token == l2Token)[0];
-    if (!bridgePoolData) throw new Error(`No bridge pool initialized for ${l2Token}`);
+  getBridgePoolForL2Token(l2Token: string, chainId: string): BridgePoolData {
+    const bridgePoolData = Object.values(this.bridgePools).filter(
```
Comment (mrice32) [inline packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts:217]: nit: find will only return one value, making this a little simpler

**Review by mrice32 — APPROVED** [review]
Looks good!

File: packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts (hunk 4 above)
Comment (nicholaspai) [reply packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts]: +1 fixed

File: packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts [hunk 5]
```diff
@@ -213,9 +213,11 @@ export class InsuredBridgeL1Client {
     return this.bridgePools[l1Token];
   }
 
-  getBridgePoolForL2Token(l2Token: string): BridgePoolData {
-    const bridgePoolData = Object.values(this.bridgePools).filter((bridgePool) => bridgePool.l2Token == l2Token)[0];
-    if (!bridgePoolData) throw new Error(`No bridge pool initialized for ${l2Token}`);
+  getBridgePoolForL2Token(l2Token: string, chainId: string): BridgePoolData {
+    const bridgePoolData = Object.values(this.bridgePools).find((bridgePool) => {
```
Comment (mrice32) [inline packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts:217]: I think you might need to remove the braces around the inner function or add a "return". Not 100% sure, but I think the current syntax will cause the inner function to always return undefined.

**Pull request merged**